operator/pkg/resources/service_account.go

```diff
@@ -90,3 +90,25 @@ func (s *ServiceAccountResource) Key() types.NamespacedName {
 	}
 	return types.NamespacedName{Name: *s.pandaCluster.Spec.ServiceAccount, Namespace: s.pandaCluster.Namespace}
 }
+
+func serviceAccountResourceKey(cluster *vectorizedv1alpha1.Cluster) types.NamespacedName {
+	if cluster.Spec.ServiceAccount == nil || *cluster.Spec.ServiceAccount == "" {
+		return types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}
+	}
+	return types.NamespacedName{Name: *cluster.Spec.ServiceAccount, Namespace: cluster.Namespace}
+}
```

Comment:
There's a very nice `ptr` library that makes these types of operations more pleasant to write. When possible, it's also preferable to not duplicate logic/declarations as it makes the intent of the code much less clear to the reader.

```suggestion
import (
	"k8s.io/utils/ptr"
)

        // Though ptr.Deref(cluster.Spec.ServiceAccount,  cluster.Name) would be preferable if we don't have to handle the "" case.
	name := ptr.Deref(cluster.Spec.ServiceAccount, "")
	if name == "" {
	    name = cluster.Name
	}
	
	return types.NamespacedName{Name: name, Namespace: cluster.Namespace}
}
```